cupy/core/core.pyx

```diff
@@ -2811,8 +2812,9 @@ cpdef ndarray tensordot_core(
             out = _ndarray_init(ret_shape, dtype)
 
     if m == 1 and n == 1:
-        _tensordot_core_mul_sum(
-            a.ravel(), b.ravel(), _manipulation._reshape(out, ()))
+        # TODO(leofang): switch back to _tensordot_core_mul_sum when
+        # ReductionKernel supports CUB block reduction
+        out = (a.ravel() * b.ravel()).sum(out=_manipulation._reshape(out, ()))
```

Comment:
It looks to degrade the performance when CUB is not enabled because your implementation calls kernels twice.

Reply:
I thought so too, but two kernel launches turned out still faster...Do you know why?

What do you think we add a for loop for routine accelerators here, and then fall back to `_tensordot_core_mul_sum`?

Reply:
@asi1024 PTAL at 1bd32b5.